src/syscalls/main.rs

```diff
@@ -0,0 +1,305 @@
+use bcc::core::BPF;
+use bcc::BccError;
+use clap::{App, Arg, ArgMatches};
+
+use core::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::{mem, ptr, thread, time};
+
+mod syscall;
+
+// A tool for reporting syscall count and latency
+//
+// Based on: https://github.com/iovisor/bcc/blob/master/tools/syscount.py
+
+#[repr(C)]
+struct data_key_t {
+    count: u64,
+    total_ns: u64,
+}
+
+const MILLISECOND: u64 = 1_000 * MICROSECOND;
+const MICROSECOND: u64 = 1_000 * NANOSECOND;
+const NANOSECOND: u64 = 1;
+
+fn get_code(matches: &ArgMatches) -> String {
+    let code = include_str!("bpf.c");
```

Comment:
Would making this `let mut code = include_str!("bpf.c").to_string();` cleanup a lot of the aliasing below?